libs-scala/resources/src/main/scala/com/digitalasset/resources/ProgramResource.scala

```diff
@@ -67,9 +67,12 @@ class ProgramResource[T](
 }
 
 object ProgramResource {
-  abstract class StartupException(message: String)
-      extends RuntimeException(message)
-      with NoStackTrace
 
-  abstract class SuppressedStartupException extends RuntimeException
+  trait StartupException extends NoStackTrace {
+    self: Exception =>
```

Comment:
Why the self-type?

Reply:
To make sure people don't forget to actually extend `RuntimeException` or similar.

Reply:
Then why removing the `RuntimeException`? :slightly_smiling_face:

Reply:
Because it allows me to do `new LedgerIdMismatchException(…) with StartupException`, which means that `LedgerIdMismatchException` can be moved to _ledger-api-common_, which doesn't have access to `StartupException` and doesn't know anything about the concept of "startup".